groovy/gemm_template.groovy

```diff
@@ -0,0 +1,220 @@
+                    if (buildStatus == hudson.model.Result.FAILURE) {
+                        sh'''
+                            echo "[FAILED] Docker image build Job URL: ${cur_job_url}/console .<br>" | tee ${WORKSPACE}/image_build.log
+                        '''
+                        throw new Exception("Docker image build job failed")
+                    }
+                    sh'''
+                        #!/usr/bin/env bash
+                        docker_image_tag=`cat ${LOG_DIR}/docker_image_tag.log`
+                        docker pull ${DOCKER_IMAGE_NAMESPACE}:${docker_image_tag}
+                    '''
+                }
+            }
+            if (fileExists("${WORKSPACE}/${LOG_DIR}/docker_image_tag.log")) {
+                archiveArtifacts  "${LOG_DIR}/docker_image_tag.log"
+            }
+        } catch (Exception e) {
+            sh'''
+                #!/usr/bin/env bash
+                set -ex
+                echo "Job URL: ${BUILD_URL}/console .<br>" | tee -a ${WORKSPACE}/image_build.log
+            '''
+            emailext(
+                subject: "Inductor TAS pipeline Pre-Check failed",
+                mimeType: "text/html",
+                from: "pytorch_inductor_val@intel.com",
+                to: "$mail_list",
+                body: '${FILE, path="image_build.log"}'
+            )
+            archiveArtifacts "image_build.log"
+            throw e
+        }
+    }
+
+    stage("prepare container"){
+        withCredentials([usernamePassword(credentialsId: 'caas_docker_hub', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]){
+            sh'''
+                #!/usr/bin/env bash
+                docker_image_tag=`cat ${LOG_DIR}/docker_image_tag.log`
+                docker pull ${DOCKER_IMAGE_NAMESPACE}:${docker_image_tag}
+            '''
+        } 
+    }
+
+    stage("benchmark") {
+        sh '''
+            #!/usr/bin/env bash
+            docker_image_tag=`cat ${LOG_DIR}/docker_image_tag.log`
+            docker run -tid --name inductor_test \
+                --privileged \
+                --env https_proxy=${https_proxy} \
+                --env http_proxy=${http_proxy} \
+                --env HF_HUB_TOKEN=$HF_TOKEN \
+                --net host --shm-size 1G \
+                -v ~/.cache:/root/.cache \
+                -v ${WORKSPACE}/${LOG_DIR}:/workspace/pytorch/${LOG_DIR} \
+                ${DOCKER_IMAGE_NAMESPACE}:${docker_image_tag}
+            docker cp scripts/modelbench/version_collect.sh inductor_test:/workspace/pytorch
+            docker exec -i inductor_test bash -c "bash version_collect.sh ${LOG_DIR}"
+            docker exec -i inductor_test bash -c "python -m pytest -v test/inductor/test_cpu_select_algorithm.py 2>&1 | tee ${LOG_DIR}/gemm_ut.log"
+        '''
+        if (fileExists("${WORKSPACE}/${LOG_DIR}/gemm_ut.log")) {
+                archiveArtifacts  "${LOG_DIR}/gemm_ut.log"
+            }
+    }

```

Comment:
Will we setup email report for this test?

Reply:
Added the Email notification.